rpc/legacyrpc/methods.go

```diff
@@ -2681,7 +2682,24 @@ func setTicketVoteBits(icmd interface{}, w *wallet.Wallet) (interface{}, error)
 		return nil, err
 	}
 
-	err = w.SetVoteBitsForTicket(ticket, cmd.VoteBits)
+	var voteBitsExt []byte
+	if cmd.VoteBitsExt != nil {
+		voteBitsExt, err = hex.DecodeString(*cmd.VoteBitsExt)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(voteBitsExt) > stake.MaxSingleBytePushLength-2 {
+			return nil, fmt.Errorf("bad ext votebits length (got %v, max %v)",
```

Comment:
So what I would do is move this check to the wallet's `SetVoteBitsForTicket` method, before the update transaction is opened.  Then it will be checked everywhere.  If you want to do an early valid parameters check in this RPC handler, that is fine as well but any errors need to be *dcrjson.RPCError with the correct error code for an invalid parameter.  Also, these errors need an explanation of *why* the value is bad.  I would also avoid abbreviations (say extended, not ext).